go/vt/vtctl/workflow/traffic_switcher.go

```diff
@@ -1203,6 +1220,278 @@ func (ts *trafficSwitcher) isSequenceParticipating(ctx context.Context) (bool, e
 	return sequenceFound, nil
 }
 
+// getTargetSequenceMetadata returns a map of sequence metadata keyed by the
+// backing sequence table name. If the target keyspace has no tables
+// defined that use sequences for auto_increment generation then a nil
+// map will be returned.
+func (ts *trafficSwitcher) getTargetSequenceMetadata(ctx context.Context) (map[string]*sequenceMetadata, error) {
+	vschema, err := ts.TopoServer().GetVSchema(ctx, ts.targetKeyspace)
+	if err != nil {
+		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "failed to get vschema for target keyspace %s: %v",
+			ts.targetKeyspace, err)
+	}
+	if vschema == nil || vschema.Tables == nil || len(vschema.Tables) == 0 { // Nothing to do
+		return nil, nil
+	}
+
+	targets := maps.Values(ts.Targets())
+	if len(targets) == 0 || targets[0].GetPrimary() == nil { // This should never happen
+		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "no primary tablet found for target keyspace %s", ts.targetKeyspace)
+	}
+	targetDBName := targets[0].GetPrimary().DbName()
+	sequencesByBackingTable := make(map[string]*sequenceMetadata)
+	smMu := sync.Mutex{}
+	for _, table := range ts.Tables() {
+		vs, ok := vschema.Tables[table]
+		if !ok || vs == nil {
+			continue
+		}
+		if vs.AutoIncrement != nil && vs.AutoIncrement.Sequence != "" {
+			sm := &sequenceMetadata{
+				backingTableName:     vs.AutoIncrement.Sequence,
+				usingTableName:       table,
+				usingTableDefinition: vs,
+				usingTableDBName:     targetDBName,
+			}
+			// If the sequence table is fully qualified in the vschema then
+			// we don't need to find it later.
+			if strings.Contains(vs.AutoIncrement.Sequence, ".") {
+				parts := strings.Split(vs.AutoIncrement.Sequence, ".")
+				if len(parts) != 2 {
+					return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "invalid sequence table name %s defined in the %s keyspace",
+						vs.AutoIncrement.Sequence, ts.targetKeyspace)
+				}
+				sm.backingTableName = parts[1]
+				sm.backingTableKeyspace = parts[0]
+			}
+			sequencesByBackingTable[sm.backingTableName] = sm
+		}
+	}
+	if len(sequencesByBackingTable) == 0 { // Nothing to do
+		return nil, nil
+	}
+
+	select {
+	case <-ctx.Done():
```

Comment:
You can shortcut this particular pattern by doing:
```go
if err := ctx.Err(); err != nil {
    return nil, err
}
```

Which will return an error if the context is done or nil if not.

Reply:
Hmmm, that's true... I'll play with this and potentially change. Thanks!

Reply:
Ah, in this case it's not the same. If the internal err group context is cancelled -- because another goroutine returned an error -- then the error is nil. So I want to return whether the internal context was cancelled or the parent one hit the timeout/deadline.